Minor update - Convert JSON array to XML examples (#127765)
Azure/azure-content: articles/logic-apps/expression-functions-reference.md

<person>
```

*Example 4*

> The xml() function expects either an object or a string containing valid XML. It does not accept a raw array as input. 

Reviewer: ```suggestion
The `xml()` function expects either an object or a string containing valid XML. The function doesn't accept a raw array as input.

```

> If your data is a JSON string, you can wrap it with the json() function to convert it into an object before passing it to xml().

Reviewer: ```suggestion
If your data is a JSON string, you can use the `json()` function to convert the string to a JSON object before you pass the result to the `xml()` function, for example:
```

```
xml(
  json('{"root":{"array":[
    { "ID": 1, "Name": "James" },
    { "ID": 2, "Name": "John" },
    { "ID": 3, "Name": "Sam" }
  ]}}')
)
```

> Suppose you have the JSON array and store in Compose1 action.

Reviewer: ```suggestion
If you have a JSON array, like the following example, you have two options.
```

```json
[
  { "ID": 1, "Name": "James" },
  { "ID": 2, "Name": "John" },
  { "ID": 3, "Name": "Sam" }
]
```

> Option 1: Using a JSON object in Compose2

Reviewer: ```suggestion
Option 1: Store the JSON array in a **Compose** action named **Compose1**. Then use the `outputs()` function to return a JSON object from **Compose1**.
```

```
{
  "root": {
    "array": @{outputs('Compose1')}
  }
}
```

> and then create XML from the JSON object.

Reviewer: The example is unclear how **Compose1** is related to **Compose2**?

```suggestion
Store the returned JSON object in another action named **Compose2**. You can then use the `xml()` and `outputs()` functions to create XML from the JSON object output from **Compose2**, for example: 
```

```
xml(outputs('Compose2'))
```

> Option 2: Using concat()

Reviewer: ```suggestion
Option 2: Store the JSON array in a **Compose** action named **Compose1**. Then use the `outputs()`, `concat()`, `json()`, and `xml()` functions to create XML from the JSON object output, for example: 
```

```
xml(
  json(
    concat(
      '{"root":{"array":',
      outputs('Compose1'),
      '}}'
    )
  )
)
```

> Options 1 and 2, along with the JSON string data, return this result:

Reviewer: Ignore Copilot recommendation and use this one instead:

```suggestion
All three examples, which include the JSON string data example, Option 1, and Option 2, return the following XML result:
```

```
xml(
  json('{"root":{"array":[
    { "ID": 1, "Name": "James" },
    { "ID": 2, "Name": "John" },
    { "ID": 3, "Name": "Sam" }
  ]}}')
)
```

If you have a JSON array, like the following example, you have two options.

```json
[
  { "ID": 1, "Name": "James" },
  { "ID": 2, "Name": "John" },
  { "ID": 3, "Name": "Sam" }
]
```

Option 1: Store the JSON array in a **Compose** action named **Compose1**. Then use the `outputs()` function to return a JSON object from **Compose1**.

```
{
  "root": {
    "array": @{outputs('Compose1')}
  }
}
```

Store the returned JSON object in another action named **Compose2**. You can then use the `xml()` and `outputs()` functions to create XML from the JSON object output from **Compose2**, for example: 

```
xml(outputs('Compose2'))
```

Option 2: Store the JSON array in a **Compose** action named **Compose1**. Then use the `outputs()`, `concat()`, `json()`, and `xml()` functions to create XML from the JSON object output, for example: 

```
xml(
  json(
    concat(
      '{"root":{"array":',
      outputs('Compose1'),
      '}}'
    )
  )
)
```

> All three examples, which include the JSON string data example, Option 1, and Option 2, return the following XML result:

Reviewer 2: ```suggestion
All three examples, which include the JSON string data example, option 1, and option 2, return the following XML result:
```
